mne/io/nirx/nirx.py

```diff
@@ -0,0 +1,204 @@
+        subject_info['first_name'] = inf['name'].split()[0].replace("\"", "")
+        subject_info['middle_name'] = inf['name'].split()[-2].replace("\"", "")
+        subject_info['birthday'] = inf['age']
+        subject_info['sex'] = inf['gender'].replace("\"", "")
+        # Recode values
+        if subject_info['sex'] in {'M', 'Male', '1'}:
+            subject_info['sex'] = 1
+        if subject_info['sex'] in {'F', 'Female', '2'}:
+            subject_info['sex'] = 2
+        # NIRStar does not record an id, or handedness by default
+
+        # Read header file
+        # This is a bit tricky as the header file isnt compliant with
+        # the config specifications. So we need to remove all text
+        # between comments before passing to config parser
+        hdr_str = open(file_hdr[0]).read()
+        hdr_str = re.sub('#.*?#', '', hdr_str, flags=re.DOTALL)
+        hdr = cp.RawConfigParser()
+        hdr.read_string(hdr_str)
+
+        # Check that the file format version is supported
+        assert (hdr['GeneralInfo']['NIRStar'] == "\"15.2\""), \
+            "Only NIRStar version 15.2 is supported"
+
+        # Parse required header fields
+
+        # Extract source-detectors requested by user
+        sources = [int(s) for s in re.findall(r'(\d)-\d:\d+',
+                   hdr['DataStructure']['S-D-Key'])]
+        detectors = [int(s) for s in re.findall(r'\d-(\d):\d+',
+                     hdr['DataStructure']['S-D-Key'])]
+        sdindex = [int(s) for s in re.findall(r'\d-\d:(\d+)',
+                   hdr['DataStructure']['S-D-Key'])]
+        assert (len(sources) == len(detectors)), \
+            "Same amount of sources and detectors required"
+        assert (len(sources) == len(sdindex)), \
+            "Same amount of sources and keys required"
+
+        # Create mne structure
+        # TODO: ch_type is currently misc as I could not find appropriate
+        #       other type, the hbo and hbr type are not relevant until the
+        #       signal has been converted
+        # TODO: nchan needs to be multiplied by two as we have two wavelengths
+        #       per sensor should the underlying type be modified to
+        #       support (wavelength x channels x data)?
+        info = create_info(len(sources) * 2,
+                           hdr['ImagingParameters']['SamplingRate'],
+                           ch_types='misc')
+        info.update({'subject_info': subject_info})
+
+        # Store the subset of sources and detectors requested by user
+        # The signals between all source-detectors are stored even if they
+        # are meaningless, the user pre specifies which combinations are
+        # meaningful
+        # TODO: Is this style of info overloading allowed?
+        #       Currently I am marking all things as fnirs_ to make easier to
+        #       find and remove if there is more appropriate storage locations
+        info['fnirs_sources'] = np.asarray(sources)
```

Comment:
also better if possible to be specific about the dtype, like `np.array(sources, float)` or `np.array(sources, int)`. Eventually these will need to be stored in FIF somewhere and how we do it might depend on the dtype that they are supposed to be.

Reply:
These should all have been resolved in 53ef36d340984983f50c7c2f3a703e301a6dd447